### ISSUE TITLE
Switching the browsing mode from normal to private or vice versa in tabs tray without selecting a tab doesn't take effect  

### ISSUE THREAD

**Issue by alinamoldovan84** [mozilla-mobile/firefox-ios#14990]
## Steps to reproduce
1. Have some normal browsing tabs opened 
2. Have some private browsing tabs opened 
3. While in private browsing open the tabs tray 
4. From tabs tray switch to normal browsing mode 
5. Close the tabs tray (without tapping a tab) 

### Expected behavior
The latest visited tab in normal browsing mod should open. Please note that this is the Safari behavior also 

### Actual behavior
The latest visited tab in private browsing opens. 

### Device & build information
* Device: 14+ (16.0.3) 
* Build version: v115 (31736) 
* First seen version: ? 
### Notes
Attachments: 

https://github.com/mozilla-mobile/firefox-ios/assets/124251579/33d3331f-db6f-4e32-8ea4-dd6f769ec266

**Comment by lmarceau** [comment]
Yeah I think we only switch mode if we specifically click on a tab. Just changing the mode in tab tray has currently no impact on the mode we're in.

**Comment by isabelrios** [comment]
I have been facing this... as a personal opinion, it is very confusing .. if I go to normal mode and click done in TabTray, even though I don't click on a tab, I would expect that next time I use the app, I'm in normal mode..

**Comment by lmarceau** [comment]
@dnarcese there was a PR opened to change this behavior, is this something we want in the app?

**Comment by dnarcese** [comment]
@nbhasin2 do you remember the technical reason we changed this behaviour? I thought it was a fix to a critical bug. Let's figure out why we made this change and that we are not breaking anything by changing it.
